test/icv.spec.js

```diff
@@ -0,0 +1,117 @@
+
+const icvAxioms = fs.readFileSync(
+  path.resolve(`${__dirname}/fixtures/issues/constraints.ttl`),
+  'utf8'
+);
+
+describe('icv', () => {
+  const database = generateDatabaseName();
+  const conn = ConnectionFactory();
+
+  const beginTx = transaction.begin.bind(null, conn, database);
+
+  beforeAll(seedDatabase(database));
+  afterAll(dropDatabase(database));
+
+  it('should add integrity constraint axioms', () =>
+    icv
+      .add(conn, database, icvAxioms, { contentType: 'text/turtle' })
+      .then(res => {
+        expect(res.status).toBe(204);
+        return icv.get(conn, database);
+      })
+      .then(res => {
+        expect(res.status).toBe(200);
+        expect(res.body.length).toBeGreaterThan(0);
+      }));
+
+  // Skipping for now, as the server isn't returning the correct results
+  it.skip('should remove integrity constraint axioms', () =>
+    icv
+      .add(conn, database, icvAxioms, { contentType: 'text/turtle' })
+      .then(() =>
+        icv.remove(conn, database, icvAxioms, { contentType: 'text/turtle' })
+      )
+      .then(res => {
+        expect(res.status).toBe(204);
+        return icv.get(conn, database);
+      })
+      .then(res => {
+        expect(res.status).toBe(200);
+        expect(res.body.length).toBe(0);
+      })
+  );
+
+  it('should clear integrity constraint axioms', () =>
+    icv
+      .add(conn, database, icvAxioms, { contentType: 'text/turtle' })
+      .then(() => icv.clear(conn, database))
+      .then(res => {
+        expect(res.status).toBe(204);
+        return icv.get(conn, database);
+      })
+      .then(res => {
+        expect(res.status).toBe(200);
+        expect(res.body.length).toBe(0);
+      }));
+
+  it('should convert constraint axioms to a SPARQL query', () =>
+    icv
+      .convert(conn, database, icvAxioms, { contentType: 'text/turtle' })
+      .then(res => {
+        expect(res.body.startsWith('SELECT')).toBe(true);
+      }));
+
+  // Server is incorrectly returning text/plain. Once that fix is live
+  // we need to update these tests to expect the boolean
+  it('should validate constraints', () =>
+    icv
+      .validate(conn, database, icvAxioms, { contentType: 'text/turtle' })
+      .then(res => {
+        expect(res.status).toBe(200);
+        // expect(res.body).toBe(false);
+        expect(res.body).toBe('false');
+      }));
+
+  it('should validate constraints in a transaction', () =>
+    beginTx()
+      .then(res => {
+        expect(res.status).toBe(200);
+        return icv.validateInTx(conn, database, res.transactionId, icvAxioms, {
+          contentType: 'text/turtle',
+        });
+      })
+      .then(res => {
+        expect(res.status).toBe(200);
+        // expect(res.body).toBe(false);
+        expect(res.body).toBe('false');
+      }));
+
+  it('should report violations', () =>
+    icv.violations(conn, database, '', {}).then(res => {
+      expect(res.status).toBe(200);
+      expect(res.body).toBeNull();
```

Comment:
Really? That's kind of weird to have a 200 and a `null` body? Is the actual result `""`?